### ISSUE TITLE
Security misconfiguration: cannot access java.io.tmpdir

### ISSUE THREAD

**Issue by LeeDr** [elastic/elasticsearch#44510]
**Elasticsearch version** (`bin/elasticsearch --version`):

**Plugins installed**: []

**JVM version** (`java -version`):

**OS version** (`uname -a` if on a Unix-like system):

**Description of the problem including expected versus actual behavior**:
I'm having a problem starting the elasticsearch master service on Windows.  I install it like;
`bin/elasticsearch-service.bat install`

and I'm getting an error in the log;
`[2019-07-16T18:02:36,685][ERROR][o.e.b.Bootstrap          ] [VAGRANT-2012-R2] Exception
java.lang.SecurityException: Security misconfiguration: cannot access java.io.tmpdir`

The problem may be caused by the fact that this Windows 2012 R2 Server VM has JDK 8 and 10 in the system PATH.  
I'm setting  `JAVA_HOME=C:\windowsInstalls\8.0.0\elasticsearch\jdk` in the system environment.
but `java -version` shows this because of the PATH (I'm fixing this);
```
java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.172-b11, mixed mode)
```

Installing the elasticsearch-service does show the correct JAVA_HOME;
```
++ bin/elasticsearch-service.bat install
Installing service      :  "elasticsearch-service-x64"
Using JAVA_HOME (64-bit):  "C:\windowsInstalls\8.0.0\elasticsearch\jdk"
-Xms1g;-Xmx1g;-XX:+UseConcMarkSweepGC;-XX:CMSInitiatingOccupancyFraction=75;-XX:+UseCMSInitiatingOccupancyOnly;-Des.networkaddress.cache.ttl=60;-Des.networkaddress.cache.negative.ttl=10;-XX:+AlwaysPreTouch;-Xss1m;-Djava.awt.headless=true;-Dfile.encoding=UTF-8;-Djna.nosys=true;-XX:-OmitStackTraceInFastThrow;-Dio.netty.noUnsafe=true;-Dio.netty.noKeySetOptimization=true;-Dio.netty.recycler.maxCapacityPerThread=0;-Dlog4j.shutdownHookEnabled=false;-Dlog4j2.disable.jmx=true;-Djava.io.tmpdir=;-XX:+HeapDumpOnOutOfMemoryError;-XX:HeapDumpPath=data;-XX:ErrorFile=logs/hs_err_pid%p.log;-Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m;-Djava.locale.providers=COMPAT;-Dio.netty.allocator.type=unpooled;-XX:MaxDirectMemorySize=536870912
The service 'elasticsearch-service-x64' has been installed.
++ net start elasticsearch-service-x64
The Elasticsearch 8.0.0-SNAPSHOT (elasticsearch-service-x64) service is starting..
The Elasticsearch 8.0.0-SNAPSHOT (elasticsearch-service-x64) service was started successfully.
```
Note that in that service install output it has `-Djava.io.tmpdir=;`

**Steps to reproduce**:

 1. download and unzip elasticsearch 8.0 (master snapshot) on Windows (FYI, there's no spaces in my elasticsearch install path)
 2. set JAVA_HOME to the jdk in the elasticsearch directory
 3. run `bin\elasticsearch-service.bat install` from an elevated (administrator) cmd prompt or git-bash shell
4. try to start the elasticsearch service
5. check the log for the error

**Provide logs (if relevant)**:

```
$ cat ../logs/elasticsearch.log
[2019-07-16T18:02:36,685][ERROR][o.e.b.Bootstrap          ] [VAGRANT-2012-R2] Exception
java.lang.SecurityException: Security misconfiguration: cannot access java.io.tmpdir
        at org.elasticsearch.bootstrap.Security.selfTest(Security.java:411) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Security.configure(Security.java:132) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:214) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:339) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124) [elasticsearch-cli-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.cli.Command.main(Command.java:90) [elasticsearch-cli-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:115) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Caused by: java.lang.SecurityException: Unable to create temporary file or directory
        at java.nio.file.TempFileHelper.create(TempFileHelper.java:142) ~[?:?]
        at java.nio.file.TempFileHelper.createTempFile(TempFileHelper.java:160) ~[?:?]
        at java.nio.file.Files.createTempFile(Files.java:915) ~[?:?]
        at org.elasticsearch.bootstrap.Security.selfTest(Security.java:404) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        ... 10 more
```

I looked here https://www.elastic.co/guide/en/elasticsearch/reference/master/es-tmpdir.html but it doesn't mention Windows

**Comment by LeeDr** [comment]
I made some changes to the install process to get rid of the other jdk versions in the path and add the elasticsearch/jdk to the path but still get the emtpy tmpdir during the service install and the error on starting the service;

```
++ java -version
openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)

++ echo 'C:\windowsInstalls\8.0.0\elasticsearch\jdk'
C:\windowsInstalls\8.0.0\elasticsearch\jdk

++ bin/elasticsearch-service.bat install
Installing service      :  "elasticsearch-service-x64"
Using JAVA_HOME (64-bit):  "C:\windowsInstalls\8.0.0\elasticsearch\jdk"
-Xms1g;-Xmx1g;-XX:+UseConcMarkSweepGC;
-XX:CMSInitiatingOccupancyFraction=75;-XX:+UseCMSInitiatingOccupancyOnly;-Des.networkaddress.cache.ttl=60;
-Des.networkaddress.cache.negative.ttl=10;-XX:+AlwaysPreTouch;-Xss1m;
-Djava.awt.headless=true;-Dfile.encoding=UTF-8;-Djna.nosys=true;
-XX:-OmitStackTraceInFastThrow;-Dio.netty.noUnsafe=true;
-Dio.netty.noKeySetOptimization=true;-Dio.netty.recycler.maxCapacityPerThread=0;-Dlog4j.shutdownHookEnabled=false;
-Dlog4j2.disable.jmx=true;-Djava.io.tmpdir=;
-XX:+HeapDumpOnOutOfMemoryError;-XX:HeapDumpPath=data;
-XX:ErrorFile=logs/hs_err_pid%p.log;
-Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m;
-Djava.locale.providers=COMPAT;-Dio.netty.allocator.type=unpooled;-XX:MaxDirectMemorySize=536870912
```

**Comment by LeeDr** [comment]
Here's some logging from 7.3.0 BC4 where I have JAVA_HOME set to the bundled jdk;

```
++ bin/elasticsearch-service.bat install
Installing service      :  "elasticsearch-service-x64"
Using JAVA_HOME (64-bit):  "C:\windowsInstalls\7.3.0\elasticsearch\jdk"
-Xms1g;-Xmx1g;-XX:+UseConcMarkSweepGC;-XX:CMSInitiatingOccupancyFraction=75;-XX:+UseCMSInitiatingOccupancyOnly;-Des.networkaddress.cache.ttl=60;-Des.networkaddress.cache.negative.ttl=10;-XX:+AlwaysPreTouch;-Xss1m;-Djava.awt.headless=true;-Dfile.encoding=UTF-8;-Djna.nosys=true;-XX:-OmitStackTraceInFastThrow;-Dio.netty.noUnsafe=true;-Dio.netty.noKeySetOptimization=true;-Dio.netty.recycler.maxCapacityPerThread=0;-Dlog4j.shutdownHookEnabled=false;-Dlog4j2.disable.jmx=true;-Djava.io.tmpdir=;-XX:+HeapDumpOnOutOfMemoryError;-XX:HeapDumpPath=data;-XX:ErrorFile=logs/hs_err_pid%p.log;-Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m;-Djava.locale.providers=COMPAT;-Dio.netty.allocator.type=unpooled;-XX:MaxDirectMemorySize=536870912
The service 'elasticsearch-service-x64' has been installed.
```
again, `-Djava.io.tmpdir=;` ^ 
Fails to start with same Security misconfiguration error.

**Comment by rjernst** [comment]
This is a bug introduced in 7.2.0 from a refactoring of how the cli tools handle tempdir. I have a fix at https://github.com/elastic/elasticsearch/pull/44541, but am having troubling getting a test that fails before the fix.

**Comment by rjernst** [comment]
Fixed by #44541
